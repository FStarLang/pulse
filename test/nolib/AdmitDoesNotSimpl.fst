module AdmitDoesNotSimpl

#lang-pulse
open Pulse.Nolib

assume val foo : int -> slprop

fn test (x y : int)
  requires foo (fst (x,y))
  ensures foo x
{
  admit(); (* Should show foo (fst (x,y)) *)
}

fn test2 (x y : int)
  requires foo (fst (x,y))
  ensures foo x
{
  assert foo x;
  admit(); (* Should show foo x *)
}

fn test3 ()
  requires foo (1+1)
  ensures foo 2
{
  admit(); (* Should show foo (1+1) *)
}

fn test4 ()
  requires foo (1+1)
  ensures foo 2
{
  ();
  admit(); (* Stil foo (1+1) *)
}
