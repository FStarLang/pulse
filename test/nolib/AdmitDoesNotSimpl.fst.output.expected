* Info at AdmitDoesNotSimpl.fst(12,2-12,9):
  - Admitting continuation.
  - Current context:
      AdmitDoesNotSimpl.foo (FStar.Pervasives.Native.fst (x, y))
  - In typing environment:
      y#130 : Prims.int,
      x#129 : Prims.int

* Info at AdmitDoesNotSimpl.fst(20,2-20,9):
  - Admitting continuation.
  - Current context:
      AdmitDoesNotSimpl.foo x
  - In typing environment:
      y#130 : Prims.int,
      x#129 : Prims.int

* Info at AdmitDoesNotSimpl.fst(27,2-27,9):
  - Admitting continuation.
  - Current context:
      AdmitDoesNotSimpl.foo (1 + 1)
  - In typing environment:
      uu___0#116 : Prims.unit

* Info at AdmitDoesNotSimpl.fst(35,2-35,9):
  - Admitting continuation.
  - Current context:
      AdmitDoesNotSimpl.foo 2
  - In typing environment:
      uu___0#116 : Prims.unit

